### PR TITLE
Update UserStoreManager with registry authorization related fixes.

### PR DESCRIPTION
**Pull request by dinusha-dilrukshi**

### CODE REVIEW

File: components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java [hunk 1]
```diff
@@ -257,7 +397,110 @@ public Properties getDefaultUserStoreProperties() {
...
+
+    @Override
+    protected RoleContext createRoleContext(String roleName) throws UserStoreException {
+        throw new UserStoreException("UserStoreManager method not supported : #createRoleContext");
+    }
+
+    //Todo: Implement doCheckExistingUser
+    @Override
+    protected boolean doCheckExistingUser(String userName) throws UserStoreException {
+        return true;
```
Comment (godwinamila) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: Why just returning true ?

File: components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java [hunk 2]
```diff
@@ -257,7 +397,110 @@ public Properties getDefaultUserStoreProperties() {
         return mapAttributes;
     }
 
+    //Todo: Implement doCheckExistingRole
+    @Override
+    protected boolean doCheckExistingRole(String roleName) throws UserStoreException {
+        return true;
```
Comment (godwinamila) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: Why just returning true ?

File: components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java [hunk 3]
```diff
@@ -340,9 +633,12 @@ public Date getPasswordExpirationTime(String userName) throws UserStoreException
             if (response == HttpStatus.SC_OK) {
                 String respStr = new String(getMethod.getResponseBody());
                 JSONObject resultObj = new JSONObject(respStr);
-                JSONArray groups = resultObj.getJSONArray("roles");
+                JSONArray groups = resultObj.getJSONArray("groups");
```
Comment (godwinamila) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: On premise agent returns roles as "roles" JSON Array. Why did we changed this groups ?

Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: Yes. This has been changed as "groups". 
Refer https://github.com/wso2-extensions/identity-agent-onprem-userstore/blob/master/components/org.wso2.carbon.identity.agent.onprem.userstore/src/main/java/org/wso2/carbon/identity/agent/onprem/userstore/resource/GroupResource.java#L80

File: components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java (hunk 2 above)
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: We need to implement this from on-premise-userstore side and call that method here. 
//Todo: Implement doCheckExistingRole

File: components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java (hunk 1 above)
Comment (dinusha-dilrukshi) [inline components/org.wso2.carbon.identity.user.store.onprem/src/main/java/org/wso2/carbon/identity/user/store/ws/WSUserStoreManager.java]: We need to implement this from on-premise-userstore side and call that method here. 
//Todo: Implement doCheckExistingUser

**Pull request merged**